### PR TITLE
Enable unlocking of local repository on NFS volume

### PR DESCRIPTION
**Pull request by hmsayem**

### CODE REVIEW

File: pkg/unlock.go [hunk 1]
```diff
@@ -95,6 +95,12 @@ func NewCmdUnlockRepository(clientGetter genericclioptions.RESTClientGetter) *co
 			if err != nil {
 				return fmt.Errorf("setup option for repository failed")
 			}
+
+			if setupOpt.Provider == v1.ProviderLocal && hostPath == "" {
+				return fmt.Errorf("host path must be specified")
+			}
```
Comment (hossainemruz) [inline pkg/unlock.go:101]: You can't enforce the `local` backend will always be `hostPath`. It support any Kubernetes volume such as NFS, PVC etc.

Comment (hmsayem) [reply pkg/unlock.go:101]: When backing up data from a Kubernetes cluster to an NFS server, the backup data is stored in a directory that is exported by the NFS server. If the NFS server is running inside the Kubernetes cluster, the exported directory will also be inside the cluster.

However, when unlocking the backup repository using the restic command with Docker, the repository does not have access to the remote directory on the NFS server. Therefore, we need to mount the NFS share to a local directory on the machine.

In this context, the term "hostpath" refers to this local mount point directory. By using the hostpath as the local mount point, we can easily access the backed-up data from the NFS share. 

Thank you for bringing up this important point. Does my explanation address your concern?

Comment (hossainemruz) [reply pkg/unlock.go:101]: >If the NFS server is running inside the Kubernetes cluster, the exported directory will also be inside the cluster.

Then what is the purpose of backup? If cluster-wide disaster happen, we will lose the backed up data too. It is practical for demo purpose.

Comment (hossainemruz) [reply pkg/unlock.go:101]: You can't magically mount NFS share in users local machine.

Comment (hossainemruz) [reply pkg/unlock.go:101]: We run NetVolAccessor Deployment in every namespace where a Repository exist with NFS/other network volume. This Deployment has access to that network volume. You have to run the unlock command inside that deployment pod. Similar to what we do for Snapshots. 
Ref: https://github.com/stashed/enterprise/blob/089b0070367a41feab48dc059933212ee3fb838a/pkg/registry/snapshot/utils.go#L135

Create an Unlock command similar to this forget command: https://github.com/stashed/enterprise/blob/089b0070367a41feab48dc059933212ee3fb838a/pkg/cmds/forget.go#L33

Then, the unlock command of cli will just invoke this new unlock command inside network volume acessor pod. Hope that make sense.

Comment (hmsayem) [reply pkg/unlock.go:101]: Thank you for the detailed explanation.

**Pull request merged**